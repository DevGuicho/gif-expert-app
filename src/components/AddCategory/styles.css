.search {
  display: flex;
  width: auto;
  margin: 10px 0;
  border: 2px solid #cbd4e1;
  border-radius: 12px;
  overflow: hidden;
}
.search__input {
  flex-grow: 1;
  padding: 10px 10px;
  border: none;
  font-size: 14px;
  outline: none;
  color: #64748b;
}
.search__button {
  background: white;
  border: none;
  padding: 0 10px;
  cursor: pointer;
  color: #64748b;
}
.search.error {
  border: 2px solid #d83232;
}
