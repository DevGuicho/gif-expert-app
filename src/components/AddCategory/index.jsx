import React, { useState } from 'react';
import { MdSearch } from 'react-icons/md';
import PropTypes from 'prop-types';

import './styles.css';

const AddCategory = ({ setCategories, categories }) => {
  const [inputValue, setInputValue] = useState({
    category: '',
  });
  const [error, setError] = useState(false);

  const { category } = inputValue;

  const handleChange = (e) => {
    setInputValue({
      ...inputValue,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (categories.includes(inputValue.category)) {
      setError(true);
      return;
    }
    setCategories((prevState) => [inputValue.category, ...prevState]);
    setError(false);
    setInputValue({ category: '' });
  };

  return (
    <form className={`search ${error && 'error'}`} onSubmit={handleSubmit}>
      <input
        className="search__input"
        type="text"
        name="category"
        id="category"
        onChange={handleChange}
        value={category}
      />
      <button className="search__button">
        <MdSearch size="25px" />
      </button>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
};

export default AddCategory;
