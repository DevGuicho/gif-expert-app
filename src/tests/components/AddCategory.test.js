import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {
  const setCategories = jest.fn();
  const categories = [];

  let wrapper = shallow(
    <AddCategory setCategories={setCategories} categories={categories} />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(
      <AddCategory setCategories={setCategories} categories={categories} />
    );
  });

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value } });
  });

  test('No debe de postear la informaciÃ³n on submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Debe llamar el setCategories y limpiar la caja de texto', () => {
    const value = 'hola mundo';

    wrapper.find('input').simulate('change', { target: { value } });

    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect(setCategories).toHaveBeenCalled();

    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
