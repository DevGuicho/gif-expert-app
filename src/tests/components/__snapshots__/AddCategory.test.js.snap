// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrarse correctamente 1`] = `
<form
  className="search false"
  onSubmit={[Function]}
>
  <input
    className="search__input"
    id="category"
    name="category"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="search__button"
  >
    <MdSearch
      size="25px"
    />
  </button>
</form>
`;
